src/UtilityFunctions.cpp

```diff
@@ -41,10 +41,19 @@ int sumGeneratedNumbers(std::unique_ptr<NumberGenerator> gen, int num)
 
 int sumNumbersFromGenerators(std::vector<GeneratorType> const & types, int num)
 {
-    // TODO: Write a loop that goes over a collection of GeneratorType values (types)
-    // and creates generator for given type (use createGenerator function),
-    // then sums up all generated values for each type and returns the sum of sums.
-    // Use sumGeneratedNumbers to sum generate values for each generator.
-    return 42;
+    int retVal = 0;
+    try
+    {
+        for (auto type : types)
+        {
+            retVal += sumGeneratedNumbers(createGenerator(type), num);
+        }
+    }
+    catch (const std::logic_error &e)
+    {
+        std::cout << e.what() << std::endl;
+        retVal = -1;
```

Comment:
preparing "magic value" of -1 is more like C language solution, where exceptions are not available.
In C++ we would like to have this exception thrown, so catching should not be done here, but on some higher levels.
When we return -1 here it can be used in further computations and the error can pass silently. Someone may even not notice that there was a problem.

Reply:
Thanks for your help, so try {} catch() shall be moved to main?

Reply:
Yes, it would be better to have it in main